godot-core/src/builtin/meta/registration/constant.rs

```diff
@@ -0,0 +1,91 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use godot_ffi as sys;
+
+use sys::interface_fn;
+
+use crate::builtin::{meta::ClassName, StringName};
+
+/// A constant named `name` with the value `value`.
+pub struct IntegerConstant {
+    name: StringName,
+    value: i64,
+}
+
+impl IntegerConstant {
+    pub fn new<T: TryInto<i64> + std::fmt::Debug + Copy>(name: StringName, value: T) -> Self {
+        Self {
+            name,
+            value: value.try_into().ok().unwrap_or_else(|| {
+                panic!("exported constant `{value:?}` must be representable as `i64`")
```

Comment:
Maybe include the name:
```suggestion
                panic!("exported constant '`{name}`' (value `{value:?}`) must be representable as `i64`")
```